build/Common.props

```diff
@@ -57,7 +57,7 @@
     <PackageReference Include="StyleCop.Analyzers" Version="$(StyleCopAnalyzersPkgVer)" Condition="'$(SkipAnalysis)'!='true'">
       <PrivateAssets>All</PrivateAssets>
     </PackageReference>
-    <PackageReference Include="Microsoft.CodeCoverage" Version="$(MicrosoftCodeCoveragePkgVer)" Condition="'$(Configuration)'=='Release'"/>
+    <PackageReference Include="Microsoft.CodeCoverage" Version="$(MicrosoftCodeCoveragePkgVer)" PrivateAssets="All" Condition="'$(Configuration)'=='Release'" />
```

Comment:
I guess we don't need a condition?

```suggestion
    <PackageReference Include="Microsoft.CodeCoverage" Version="$(MicrosoftCodeCoveragePkgVer)" PrivateAssets="All" />
```

Reply:
Well, @eddynaka wanted it back...

Reply:
I want to understand why.

Reply:
Please find my comment here https://github.com/open-telemetry/opentelemetry-dotnet/pull/2237#discussion_r684391603.

Reply:
I agree with @reyang that Debug and Release build should have the same dependencies.